Documentation/contributing/development/contributing_guide.rst

```diff
@@ -183,13 +215,89 @@ Getting a pull request merged
 
 #. Owners of the repository will automatically adjust the labels on the pull
    request to track its state and progress towards merging.
+
 #. Once the PR has been reviewed and the CI tests have passed, the PR will be
    merged by one of the repository owners. In case this does not happen, ping
    us on Slack.
 
-
-Pull request review process
----------------------------
+#. If reviewers have requested changes and once those changes are done,
+   re-request a review for the reviewers that have requested changes. Otherwise,
+   those reviewers will not be notified and your PR will not receive any
+   reviews. If the PR is considerably large (e.g. with more than 200 lines
+   changed and/or more than 6 commits) create new commit for each review. This
+   will make the review process smoother as GitHub has limitations that
+   prevents reviewers from only seeing the new changes added since the last time
+   they have reviewed a PR. Once all reviews are addressed those commits should
+   be squashed against the commit that introduced those changes. This can be
+   easily accomplished by the usage of ``git rebase -i origin/master`` and in
+   that windows, move these new commits below the commit that introduced the
+   changes and replace the work ``pick`` with ``fixup``. In the following
+   example, commit ``d2cb02265`` will be meld into ``9c62e62d8`` and commit
+   ``146829b59`` will be meld into ``9400fed20``.
+
+       ::
+
+           pick 9c62e62d8 docs: updating contribution guide process
+           fixup d2cb02265 joe + paul + chris changes
+           pick 9400fed20 docs: fixing typo
+           fixup 146829b59 Quetin and Maciej reviews
+
+   Once this is done you can perform
+   ``git push origin <branch-name> --force`` to push your changes
+   into your PR and request for your PR to be merged.
+
+
+Pull requests review process for committers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. Every committer in the `committers team <https://github.com/orgs/cilium/teams/committers/members>`_
+   belongs to `one or more other teams in the Cilium organization <https://github.com/orgs/cilium/teams/team/teams>`_
+   if you would like to be added or removed from any team, please contact any
+   of the `maintainers <https://github.com/orgs/cilium/teams/maintainers/members>`_.
+
+#. Once a PR is open, GitHub will automatically pick which `teams <https://github.com/orgs/cilium/teams/team/teams>`_
+   should review the PR using the ``CODEOWNERS`` file. Each committer can see
+   PRs they need to review by filtering PRs they has been requested reviews for.
+   A good filter is provided in this `link <https://github.com/cilium/cilium/pulls?q=is%3Apr+is%3Aopen+draft%3Afalse+review-requested%3A%40me+sort%3Aupdated-asc>`_
+   so make sure to bookmark it.
```

Comment:
```suggestion
   should review the PR using the ``CODEOWNERS`` file. Each committer can see the
   PRs they need to review by filtering by reviews requested.
   A good filter is provided in this `link <https://github.com/cilium/cilium/pulls?q=is%3Apr+is%3Aopen+draft%3Afalse+review-requested%3A%40me+sort%3Aupdated-asc>`_
   so make sure to bookmark it.
```

Hopefully this simplifies it. I had a hard time parsing the previous sentence. Feel free to modify.